docker/run_on_docker.sh

```diff
@@ -85,23 +85,34 @@ bootstrap_image() {
   rm -fR $build_workdir
 }
 
+push_image() {
+    local image_name="$1"
+    docker push $image_name || die "Failed to push docker image $image_name"
```

Comment:
Perhaps we can add a mode where failure to push is only a warning rather than a fatal error.  This would allow us continue to operate even if the dockerhub machine is down.